model/src/main/java/com/javahelp/model/userEntityTest/User.java

```diff
@@ -0,0 +1,54 @@
+package com.javahelp.model.userEntityTest;
+import com.javahelp.model.user.UserInfo;
+import com.javahelp.model.user.ClientUserInfo;
+import com.javahelp.model.user.UserType;
+import com.javahelp.model.user.ProviderUserInfo;
+
+public class User {
+    /**
+     * A class for a user entity
+     */
+
+    private String stringID;
+    private UserInfo userInfo;
```

Comment:
We may also want a username or email field here. This would be unique, like ID, but unlike ID, would not be immutable.

Reply:
Yeah, I think username (or the user's first name and last name) as well as their email are stored inside UserInfo userInfo. Is it necessary to extract them out explicitly, though?

Reply:
My line of thought is that the username and/or email really has nothing to do with the personal information of each client or provider, and is really more administrative. If it's something like a username, I think it probably belongs in user, but for I guess maybe for something like an email it makes more sense in UserInfo.